### PR TITLE
GCR & ACR guide

### PR DESCRIPTION
**Pull request by dchw**
GCP and Azure articles

### CODE REVIEW

File: docs/SUMMARY.md [hunk 1]
```diff
@@ -16,6 +16,7 @@
     * [Multi-platform builds](guides/multi-platform.md)
     * [CI integration](guides/ci-integration.md)
     * [Pushing and Pulling from AWS ECR](guides/pushing-images-to-ECR.md)
+    * [Pushing and Pulling Images with GCP Artifact Registry](./pushing-images-to-GCR.md)
```
Comment (vladaionescu) [inline docs/SUMMARY.md]:
```suggestion
    * [Pushing and Pulling Images with GCP Artifact Registry](guides/pushing-images-to-GCR.md)
```

File: docs/guides/pushing-images-to-GCR.md [hunk 2]
```diff
@@ -0,0 +1,97 @@
+# Pushing and Pulling images with GCP Artifact Registry
+
+## Introduction
+
+The GCP Artifact Registry is is a hosted docker repository that requires extra configuration for day-to-day use. This configuration is not typical of other repositories, and there are some considerations to account for when using it with Earthly. This guide will walk you through creating an Earthfile, building an image, and pushing it to Artifact Registry.
```
Comment (vladaionescu) [inline docs/guides/pushing-images-to-GCR.md:5]:
```suggestion
The GCP Artifact Registry is a hosted docker repository that requires extra configuration for day-to-day use. This configuration is not typical of other repositories, and there are some considerations to account for when using it with Earthly. This guide will walk you through creating an Earthfile, building an image, and pushing it to Artifact Registry.
```

**Review by vladaionescu — APPROVED** [review]

File: docs/SUMMARY.md (hunk 1 above)
Comment (dchw) [reply docs/SUMMARY.md]: Already fixed

**Comment by dchw** [comment]
@vladaionescu Re-requested review to look at the Azure article

File: docs/SUMMARY.md [hunk 3]
```diff
@@ -15,7 +15,10 @@
     * [Debugging techniques](guides/debugging.md)
     * [Multi-platform builds](guides/multi-platform.md)
     * [CI integration](guides/ci-integration.md)
-    * [Pushing and Pulling from AWS ECR](guides/pushing-images-to-ECR.md)
+    * Working With Cloud Providers
+        * [Pushing and Pulling from AWS ECR](guides/pushing-images-to-ECR.md)
+        * [Pushing and Pulling Images with GCP Artifact Registry](guides/pushing-images-to-GCR.md)
+        * [Pushing and Pulling Images with Azure ACR](guides/pushing-images-to-ACR.md)
```
Comment (vladaionescu) [inline docs/SUMMARY.md]: Shorter names look better here - how about just:

```suggestion
    * Working With Cloud Providers
        * [AWS ECR](guides/pushing-images-to-ECR.md)
        * [GCP Artifact Registry](guides/pushing-images-to-GCR.md)
        * [Azure ACR](guides/pushing-images-to-ACR.md)
```

File: docs/guides/pushing-images-to-ACR.md [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+	},
+	"credsStore": "acr-linux"
+}
+```
+
+ACR boasts many other methods of logging in, including [Service Principals](https://docs.microsoft.com/en-us/azure/container-registry/container-registry-auth-service-principal) and [admin accounts](https://docs.microsoft.com/en-us/azure/container-registry/container-registry-authentication#admin-account). Note that the admin account method is not recommended for production usage. Please follow the releavant guides to authenticate if you wish to use one of these other methods.
+
+## RBAC
+
+Ensure that you have correct permissions to push and pull the images. Please reference the [ACR RBAC documentation](https://docs.microsoft.com/en-us/azure/container-registry/container-registry-roles) to ensure you have the correct permissions set. To complete all the activities in this guide, you wull need to have at least the permissions detailed in the built-in `Contributor` role.
```
Comment (vladaionescu) [inline docs/guides/pushing-images-to-ACR.md:49]:
```suggestion
Ensure that you have correct permissions to push and pull the images. Please reference the [ACR RBAC documentation](https://docs.microsoft.com/en-us/azure/container-registry/container-registry-roles) to ensure you have the correct permissions set. To complete all the activities in this guide, you will need to have at least the permissions detailed in the built-in `Contributor` role.
```

File: docs/guides/pushing-images-to-ACR.md [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+
+## Run the Target
+
+Once we are logged in, and have the optional credential helper installed, there are no no special things left to do. To build and push an image, simply execute the build target. Don't forget the `--push` flag!
```
Comment (vladaionescu) [inline docs/guides/pushing-images-to-ACR.md:53]:
```suggestion
Once you are logged in, and have the optional credential helper installed, there you are ready to use Earthly to access images in ACR. To build and push an image, simply execute the build target. Don't forget the `--push` flag!
```

**Review by vladaionescu — APPROVED** [review]
:+1:

File: docs/SUMMARY.md (hunk 3 above)
Comment (dchw) [reply docs/SUMMARY.md]: Check my latest push, I pruned that up

Comment (dchw) [reply docs/SUMMARY.md]: But had lengthened the upper one. These guides arent quite wide enough to warrant a "Working With Cloud Providers"

**Pull request merged**